pkg/controller/attach_detach.go

```diff
@@ -741,45 +842,50 @@ func (r *ReconcileAttachDetach) recoverAzVolumeAttachment(ctx context.Context, r
 		wg.Add(1)
 		go func(azv azdiskv1beta2.AzVolumeAttachment, azvMap *sync.Map) {
 			defer wg.Done()
+			var publishCtx map[string]string
+			if lun, _, err := r.cloudDiskAttacher.GetDiskLun(azv.Spec.VolumeName, azv.Spec.VolumeID, types.NodeName(azv.Spec.NodeName)); err == nil {
```

Comment:
Hi @sunpa93 Could you share the code links of `managedDiskController` and `disk.DiskProperties.ShareInfo` ? Thanks.

Reply:
This is code block on getting the disk with the disk client.
https://github.com/edreed/cloud-provider-azure/blob/9397319d763831f7e9a1d8aa8253f23d15d558cc/pkg/provider/azure_managedDiskController.go#L324

And this is what `compute.Disk` struct looks like:
https://github.com/edreed/cloud-provider-azure/blob/9397319d763831f7e9a1d8aa8253f23d15d558cc/vendor/github.com/Azure/azure-sdk-for-go/services/compute/mgmt/2022-03-01/compute/models.go#L4078-:4100

`disk.ManagedByExtended` and `disk.Properties.ShareInfos` reflect on the vms the disk is attached to. Now that I think about it, it makes more sense to rely on `disk.ManagedByExtended` instead.

Reply:
Hi @sunpa93 I found the description of `disk.ManagedByExtended` here:  https://learn.microsoft.com/en-us/rest/api/compute/disks/get?tabs=HTTP#disk. So I reached out to Chetan Maddula from Disk RP team and confirmed that we can either use `disk.ManagedByExtended` or `disk.DiskProperties.DiskState` since they are linked data as a source of truth (unless there are bugs between crp and diskrp).

Next step: I'll run some examples to see how the property would change from shared disks vs non shared disks to confirm that we can rely on that to recover our AzVolumeAttachment.

Reply:
Awesome! Great to hear that there is a source of truth we can safely rely on. Thanks for checking with the DiskRP team to confirm this. Good progress indeed!